src/convert_image/avx2.rs

```diff
@@ -840,6 +841,147 @@ unsafe fn lrgb_to_i420_avx2(
     true
 }
 
+#[inline(always)]
+fn rgb_to_bgra(
+    width: u32,
+    height: u32,
+    last_src_plane: u32,
+    src_strides: &[usize],
+    src_buffers: &[&[u8]],
+    last_dst_plane: u32,
+    dst_strides: &[usize],
+    dst_buffers: &mut [&mut [u8]],
+) -> bool {
+    unsafe {
+        rgb_to_bgra_avx2(
+            width,
+            height,
+            last_src_plane,
+            src_strides,
+            src_buffers,
+            last_dst_plane,
+            dst_strides,
+            dst_buffers,
+        )
+    }
+}
+
+#[inline]
+#[target_feature(enable = "avx2")]
+unsafe fn rgb_to_bgra_avx2(
+    width: u32,
+    height: u32,
+    _last_src_plane: u32,
+    src_strides: &[usize],
+    src_buffers: &[&[u8]],
+    _last_dst_plane: u32,
+    dst_strides: &[usize],
+    dst_buffers: &mut [&mut [u8]],
+) -> bool {
+    if width == 0 || height == 0 {
+        return true;
+    }
+
+    if dst_buffers.is_empty()
+        || dst_strides.is_empty()
+        || src_buffers.is_empty()
+        || src_strides.is_empty()
+    {
+        return false;
+    }
+
+    let max_stride = usize::max_value() / height as usize;
+    if (src_strides[0] > max_stride) || (dst_strides[0] > max_stride) {
+        return false;
+    }
+
+    if src_strides[0] * height as usize > src_buffers[0].len()
+        || dst_strides[0] * height as usize > dst_buffers[0].len()
+    {
+        return false;
+    }
+
+    const ALPHAS_MASK: [u8; 32] = [
+        0x00, 0x00, 0x00, 0xff, 0x00, 0x00, 0x00, 0xff, 0x00, 0x00, 0x00, 0xff, 0x00, 0x00, 0x00,
```

Comment:
Since the pattern is repeating, you can just use `_mm256_set1_epi32(DEFAULT_ALPHA as i32)` and drop this array